### PR TITLE
[#3555] Device Registry JDBC native build

### PR DESCRIPTION
**Pull request by harism**
Changes needed for device registry JDBC to make native-image build succeeds and native-image integration tests run passes.

### CODE REVIEW

File: services/base-jdbc/src/main/resources/META-INF/native-image/org/eclipse/hono/hono-service-base-jdbc/resources-config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      { "pattern":  "org\\/eclipse\\/hono\\/service\\/base\\/jdbc\\/store\\/devcon\\/base.postgresql.sql.yaml" },
+      { "pattern":  "org\\/eclipse\\/hono\\/service\\/base\\/jdbc\\/store\\/devcon\\/base.sql.yaml" },
+      { "pattern":  "org\\/eclipse\\/hono\\/service\\/base\\/jdbc\\/store\\/device\\/base.h2.sql.yaml" },
+      { "pattern":  "org\\/eclipse\\/hono\\/service\\/base\\/jdbc\\/store\\/device\\/base.postgresql.sql.yaml" },
+      { "pattern":  "org\\/eclipse\\/hono\\/service\\/base\\/jdbc\\/store\\/device\\/base.sql.yaml" },
+      { "pattern":  "org\\/eclipse\\/hono\\/service\\/base\\/jdbc\\/store\\/tenant\\/base.sql.yaml" },
+      { "pattern":  "sql\\/h2\\/01-create.tenants.sql" },
+      { "pattern":  "sql\\/h2\\/02-create.devices.sql" },
+      { "pattern":  "sql\\/postgresql\\/01-create.tenants.sql" },
+      { "pattern":  "sql\\/postgresql\\/02-create.devices.sql" }
```
Comment (sophokles73) [inline services/base-jdbc/src/main/resources/META-INF/native-image/org/eclipse/hono/hono-service-base-jdbc/resources-config.json:14]: maybe we can make these more generic, e.g. using wildcards that expand to all `yaml` and `sql` files?

File: .github/workflows/native-images-tests.yml [hunk 2]
```diff
@@ -47,17 +47,19 @@ jobs:
         -Pbuild-native-image,metrics-prometheus -am \
         -pl :hono-service-auth,:hono-service-command-router,:hono-service-device-registry-mongodb,\
         :hono-adapter-mqtt,:hono-adapter-amqp,:hono-adapter-coap,:hono-adapter-http,:hono-adapter-lora,\
-        :hono-tests
+        :hono-tests,:hono-service-device-registry-jdbc
```
Comment (sophokles73) [inline .github/workflows/native-images-tests.yml:50]: would you mind putting this after `:hono-service-device-registry-mongodb`?

File: services/base-jdbc/src/main/resources/META-INF/native-image/org/eclipse/hono/hono-service-base-jdbc/resources-config.json (hunk 1 above)
Comment (harism) [reply services/base-jdbc/src/main/resources/META-INF/native-image/org/eclipse/hono/hono-service-base-jdbc/resources-config.json]: Done, I kept some base resource directory structure under which *.yaml and *.sql files are filtered in.

File: .github/workflows/native-images-tests.yml (hunk 2 above)
Comment (harism) [reply .github/workflows/native-images-tests.yml]: Done.

**Review by sophokles73 — APPROVED** [review]
LGTM

Thanks for contributing :+1:

**Pull request merged**